lib/ini.js

```diff
@@ -108,13 +117,19 @@ const decode = (str, opt = {}) => {
   const out = Object.create(null)
   let p = out
   let section = null
+  let lineCommentArray = []
+  const commentsRegEx = /^[#;]{1,}.*$/
   //          section          |key      = value
   const re = /^\[([^\]]*)\]\s*$|^([^=]+)(=(.*))?$/i
   const lines = str.split(/[\r\n]+/g)
   const duplicates = {}
 
   for (const line of lines) {
     if (!line || line.match(/^\s*[;#]/) || line.match(/^\s*$/)) {
+      if (line && line.match(commentsRegEx)) {
```

Comment:
This if block does not need to be nested within its parent. I think it would be better to have a block that checks for empty lines and another that checks for comments.